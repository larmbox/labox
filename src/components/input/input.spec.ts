import { mount } from '@vue/test-utils';
import { expect, test } from 'vitest';
import { LInput } from '~/index';

test('mount component', () => {
  expect(LInput).toBeTruthy();

  const component = mount(LInput, {
    props: {
      id: 'id',
    },
  });

  expect(component.html()).toMatchSnapshot();
  expect(component.html()).toContain('input');
  expect(component.html()).toContain('form-field');
});

test('register css variables', () => {
  mount(LInput);
  expect(
    document.getElementById('lx-theme-provider-global')!.innerText
  ).toContain('--lx-input');
});

test('inherit attributes', () => {
  const component = mount(LInput, {
    props: { notaprop: '1', variant: 'primary' },
  });
  expect(component.html()).toContain('notaprop="1"');
  expect(component.html()).not.toContain('variant="primary"');
});
