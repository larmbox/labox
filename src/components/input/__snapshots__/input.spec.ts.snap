// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div class=\\"form-control form-field input input-md input-primary\\">
  <!--v-if-->
  <!--v-if-->
  <div class=\\"input-inner\\">
    <!--v-if-->
    <div class=\\"input-element\\"><input id=\\"id\\" type=\\"text\\" /></div>
    <!--v-if-->
  </div>
  <!--v-if-->
  <!--v-if-->
</div>"
`;
